instance/instance.go

```diff
@@ -87,50 +88,47 @@ func (s *Store) GetList(w http.ResponseWriter, r *http.Request, limit int, offse
 	return results, totalCount, nil
 }
 
-//Get a single instance by id
+// Get a single instance by id
 func (s *Store) Get(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	vars := mux.Vars(r)
 	instanceID := vars["instance_id"]
+	eTag := getIfMatch(r)
 	logData := log.Data{"instance_id": instanceID}
 
 	log.Event(ctx, "get instance", log.INFO, logData)
 
-	b, err := func() ([]byte, error) {
-		instance, err := s.GetInstance(instanceID)
-		if err != nil {
-			log.Event(ctx, "get instance: failed to retrieve instance", log.ERROR, log.Error(err), logData)
-			return nil, err
-		}
-
-		log.Event(ctx, "get instance: checking instance state", log.INFO, logData)
-		// Early return if instance state is invalid
-		if err = models.CheckState("instance", instance.State); err != nil {
-			logData["state"] = instance.State
-			log.Event(ctx, "get instance: instance has an invalid state", log.ERROR, log.Error(err), logData)
-			return nil, err
-		}
-
-		log.Event(ctx, "get instance: marshalling instance json", log.INFO, logData)
-		b, err := json.Marshal(instance)
-		if err != nil {
-			log.Event(ctx, "get instance: failed to marshal instance to json", log.ERROR, log.Error(err), logData)
-			return nil, err
-		}
+	instance, err := s.GetInstance(instanceID, eTag)
```

Comment:
Does this need wrapping with :
AcquireInstanceLock()
etc as done in dimension/dimension.go ... about line 45'ish ?

Reply:
Here, we only want to read the instance, and if the eTag has changed, then GetInstance will fail. I think we don't need to use the lock in this case.

Reply:
Update: due to other calls requiring the lock, I will need to also acquire it here, even if this call itself wouldn't need it, to prevent concurrency issues with them.